prime-router/src/main/kotlin/fhirengine/engine/CustomTranslationFunctions.kt

```diff
@@ -4,11 +4,11 @@ import gov.cdc.prime.router.Hl7Configuration
 import gov.cdc.prime.router.Receiver
 import gov.cdc.prime.router.common.DateUtilities
 import gov.cdc.prime.router.fhirengine.translation.hl7.utils.CustomContext
-import gov.cdc.prime.router.fhirengine.translation.hl7.utils.TranslationFunctions
+import gov.cdc.prime.router.fhirengine.translation.hl7.utils.Hl7TranslationFunctions
 import org.hl7.fhir.r4.model.BaseDateTimeType
 import java.time.ZoneId
 
-class CustomTranslationFunctions : TranslationFunctions {
+class CustomTranslationFunctions : Hl7TranslationFunctions() {
```

Comment:
I changed the hierarchy a bit to account for the following:
1. I wanted to add `truncateHL7Field` function to the interface. It requires a class dependency which made it difficult to include add as an implementation on the interface.
2. I only wanted a single implementation of the `truncateHL7Field` function (didn't need to duplicate it for `CustomTranslationFunctions`)
3. `CustomTranslationFunctions` requires the default implementation of the `convertDateTimeToHL7` function

I made this hierarchy that handles those issues
![Screenshot 2023-08-28 at 4 09 16 PM](https://github.com/CDCgov/prime-reportstream/assets/1209453/8fa2f40b-9b65-481d-9aa9-c3369e4536c3)